components/function-controller/internal/controllers/serverless/system_state.go

```diff
@@ -481,6 +476,21 @@ func (s *systemState) buildDeployment(cfg buildDeploymentArgs) appsv1.Deployment
 	}
 }
 
+func (s *systemState) getReplicas(defaultVal int32) *int32 {
+	if s.instance.Spec.ScaleConfig != nil {
+		// TODO: question - should it looks like this?
+		// Maybe better way is to return s.deployments.Items[0].Spec.Replicas?
+		// this resource is managed by HPA and returning always minReplicas is confusing
```

Comment:
What do you think about it? I think assigning always minReplicas to the deployment would cause problems when HPA scales up deployment (change deplo.spec.replicas to higher value) and at the same time fun-ctrl would always detect change on the deployment and at the same time scale down it (because of this template issue)

Reply:
Return replicas from deployment from cluster

Reply:
This topic is more complicated than I expected. I moved it to the separate issue:
https://github.com/kyma-project/kyma/issues/15152